pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/ServerReloadJobStatusCache.java

```diff
@@ -35,26 +42,29 @@
  *
  * <p>Thread-safe for concurrent access. Uses Guava Cache with LRU eviction
  * and time-based expiration.
+ *
+ * <p>Implements PinotClusterConfigChangeListener to support dynamic configuration
+ * updates from ZooKeeper cluster config. When config changes, cache is rebuilt
+ * with new settings and existing entries are migrated.
  */
 @ThreadSafe
-public class ServerReloadJobStatusCache {
+public class ServerReloadJobStatusCache implements PinotClusterConfigChangeListener {
   private static final Logger LOG = LoggerFactory.getLogger(ServerReloadJobStatusCache.class);
+  private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+  static final String CONFIG_PREFIX = "pinot.server.table.reload.status.cache";
```

Comment:
shall we put this into CommonConstants?

Reply:
@xiangfu0 
The reason I did not do that right now is because it is only used in this class. I generally prefer to move constants to central utils once they are used in at least 2 or more places. totally fine with either. wdyt?